lib/lazy-result.js

```diff
@@ -109,7 +109,15 @@ class LazyResult {
     this.result = new Result(processor, root, opts)
     this.helpers = { ...postcss, result: this.result, postcss }
     this.plugins = this.processor.plugins.map(plugin => {
-      if (typeof plugin === 'object' && plugin.prepare) {
+      if (
+        typeof plugin === 'function' &&
+        plugin.prototype &&
+        plugin.prototype.constructor === plugin &&
+        plugin.postcss === true
+      ) {
+        // eslint-disable-next-line new-cap
+        return new plugin(this.result)
+      } else if (typeof plugin === 'object' && plugin.prepare) {
```

Comment:
I left this line intact, but I think it would not hurt to check the type of `plugin.prepare`.

```suggestion
      } else if (typeof plugin === 'object' && typeof plugin.prepare === 'function') {
```